src/scanpy/_utils/__init__.py

```diff
@@ -247,14 +247,18 @@ def _check_array_function_arguments(**kwargs):
         )
 
 
-def _check_use_raw(adata: AnnData, use_raw: None | bool) -> bool:
+def _check_use_raw(
+    adata: AnnData, use_raw: None | bool, *, layer: str | None = None
+) -> bool:
     """
     Normalize checking `use_raw`.
 
     My intentention here is to also provide a single place to throw a deprecation warning from in future.
     """
```

Comment:
Shouldn’t it be like this? Otherwise one overrides the other.

```suggestion
    """
    if use_raw is not None and layer is not None:
        msg = "Cannot use both `use_raw` and `layer`."
        raise ValueError(msg)
```

Reply:
OK, that’s checked later